### PR TITLE
Add CA2252 benchmark

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: src/PerformanceTests/Tests/Enabled/CSharp_CA2252.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        where T : {name}_Foo, ICloneable
+    {{
+        return true;
+    }}
+}}
+
+[RequiresPreviewFeatures(""Lib is in preview."", Url = ""https://aka.ms/aspnet/kestrel/http3reqs"")]
+public class {name}_Foo
+{{
+}}
```
Comment (Youssef1313) [inline src/PerformanceTests/Tests/Enabled/CSharp_CA2252.cs:43]: I may extend it to a code that reports multiple different rules in a follow-up PR.

**Review by Youssef1313 — COMMENTED** [review]
@buyaa-n @jeffhandley Can you review this small PR?

**Review by Youssef1313 — COMMENTED** [review]
The benchmark is crashing. Will try to understand why.

**Review by Youssef1313 — COMMENTED** [review]
@jmarolf I'm updating performance tests to use a non-preview .NET 6.0

**Review by buyaa-n — APPROVED** [review]
Perf test LGTM, will wait for @jmarolf's review for infra part before merging

**Comment by Youssef1313** [comment]
@jmarolf Can you review please? Thanks!

**Review by jmarolf — APPROVED** [review]

**Pull request merged**